File: TemporaryMqLibrary/src/main/java/cattt.temporary.mq/MqConfigure.java
package cattt.temporary.mq;

import android.support.annotation.IntRange;

import cattt.temporary.mq.base.enums.QosClub;
import cattt.temporary.mq.base.enums.QosType;

public class MqConfigure {
    public static String userName = "guest";
    public static String password = "guest";

    public static String clientId = "11";
    public static String serverUri = new StringBuffer().append("tcp").append("://")
            .append("39.106.117.155").append(":").append("1883").toString();

    /**
     * key is topic
     * value is qos
     */
    public static String[] topics = null;
    @QosClub
    public static int qos = QosType.NORMAL;

    public static boolean isTrace = true;

    public static int connectionTimeout = 10;
    public static int keepAliveInterval = 60;
    @IntRange(from = 10, to = 1000)
    public static int maxInflight = 100;
    @IntRange(from = 5000, to = Integer.MAX_VALUE)
    public static int bufferSize = 10000;
}

File: CattMqKtLibs/src/main/java/catt.kt.libs.mq/MqConfigure.kt
package catt.kt.libs.mq

import android.annotation.SuppressLint

@SuppressLint("StaticFieldLeak")
object MqConfigure {
    /**
     * 需要初始化
     */
    @JvmStatic
    var deviceNo: String = ""

    /**
     * Mqtt client id
     * 连接前需要初始化
     */
    @JvmStatic
    var clientId: String = ""

    /**
     * Mqtt subscribe topics
     * 连接前需要初始化
     */
    @JvmStatic
    var topics: Array<String> = arrayOf()

    @JvmStatic
    var userName: String = ""

    @JvmStatic
    var password: String = ""

    @JvmStatic
    var serverUrl: String = ""

    /**
     * 0 服务质量普通
     * 1 服务质量高
     * 2 服务质量最高
     */
    @JvmStatic
    var qos: Int = 1

    /**
     * 是否开启MQ日志
     */
    @JvmStatic
    var isTrace: Boolean = true

    /**
     * 超时时间,单位是秒(example: 30s)
     */
    @JvmStatic
    var connectionTimeout: Int = 10
    /**
     * 保持连接时间周期,单位是秒(example: 60s)
     */
    @JvmStatic
    var keepAliveInterval: Int = 60

    /**
     * 后台是否保持连接常态化
     * true 不保持， false 保持
     */
    @JvmStatic
    var isCleanSession: Boolean = false

    /**
     * 最大容量
     * PS:高流量传输中需要增大此值
     */
    @JvmStatic
    var maxInflight: Int = 1000

    @JvmStatic
    var bufferSize: Int = Integer.MAX_VALUE
}